acceptance_modelisation/bkg_collection.py

```diff
@@ -17,10 +17,23 @@ def __init__(self, bkg_dict: dict[float, BackgroundIRF] = None):
             bkg_dict : dict of gammapy.irf.BackgroundIRF
                 The collection of model in a dictionary with as key the zenith angle (in degree) associated to the model
         """
+        bkg_dict = bkg_dict or {}
         self.bkg_dict = {}
-        if not bkg_dict is None:
-            for k in bkg_dict.keys():
-                self[k] = bkg_dict[k]
+        for k, v in bkg_dict.items():
+            key = float(k)
```

Comment:
If the conversion to float fail for some reason, it could lead to an exception a bit hard for the user to interpret in the overall context. 
Catching the exception (ValueError) and re-throwing it with a more detail message could help.